src/backend/api/meals.js

```diff
@@ -1,15 +1,306 @@
+  } catch (error) {
+    throw error;
+  }
+});
+
+// //method 2 of post
+router.post("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const meals = await knex("meal");
+    const postedMeal = meals.insert(
+      {
+      title: request.body.title,
+      description: request.body.description,
+      location: request.body.location,
+      when: request.body.when,
+      max_reservations: request.body.max_reservations,
+      price: request.body.price,
+      created_date: request.body.created_date
+    }
+    )
+    response.json(postedMeal);
   } catch (error) {
     throw error;
   }
 });
 
+// put method 1
+router.put("/:id", async (request, response) => {
+  try {
+    const meals = await knex("meal").select("*");
+    const inputId = Number(request.params.id);
+    const maxIdOfMeal =Math.max( meals.map(meal => meal.id));
+    if (isNaN(inputId)) {
+      response.send("Id is not number");
+    }
+    else if (inputId > maxIdOfMeal) {
+      response.send(`the largest id is : ${maxIdOfMeal}`)
+    }
+    else {
+      const updatedMeal = meals
+        .where({ id: request.params.id })
+        .update({
+          title: "salad",
+          description: "danlish way of salad",
+          location: "Lyngby",
+          when: "2022-01-29",
+          max_reservations: 50,
+          price: 68,
+          created_date: "2021-01-01"
+        });
+      response.json(updatedMeal);
+    }
+  } catch (error) {
+    throw error;
+  }
+});
+//method 2  
+router.put("/:id", async (request, response) => {
+  try {
+    const meals = await knex("meal").select("*");
+    const inputId = Number(request.params.id);
+    const maxIdOfMeal =Math.max( meals.map(meal => meal.id));
+    if (isNaN(inputId)) {
+      response.send("Id is not number");
+    }
+    else if (inputId >maxIdOfMeal) {
+      response.send(`the largest id is : ${maxIdOfMeal}`)
+    }
+    else {
+      const updatedMeal = meals
+        .where({ id: inputId })
+        .update(request.body.title);
+      response.json(updatedMeal);
+    }
+  } catch (error) {
+    throw error;
+  }
+});
+
+// delete
+router.delete("/:id", async (request, response) => {
+  try {
+    const meals = await knex("meal").select("*");
+    const inputId = Number(request.params.id);
+    const maxIdOfMeal =Math.max( meals.map(meal => meal.id));
+    if (isNaN(inputId)) {
+      response.send("Id is not number");
+    }
+    else if (inputId > maxIdOfMeal) {
+      response.send(`the largest id is : ${maxIdOfMeal}`)
+    }
+    else {
+      const deletedMeal = meals
+        .where({ id: request.params.id })
+        .delete();
+      response.json(deletedMeal);
+    }
+  } catch (error) {
+    throw error;
+  }
+});
+
+
+//GET api/meals/  Query parameters 
+
+router.get("/", async (request, response) => {
```

Comment:
While it works as expected there a few things that could be done better and if you're up for it you can improve it:
- The route will not work with multiple parameters at the same time as expected. If for instance I pass in both 'maxPrice' and 'availableReservations' you will return meals filtered only by the MaxPrice parameters since that is the one you check first.
- Check one more comment below for the isNaN check

I strongly suggest you watch the following [video ](https://www.youtube.com/watch?v=4KV79W4g2QU&t=418s)made by Orhan at HYF which explains how you can achieve the optimal behavior with the multiple params.

Reply:
this video really helps . It is very clear. Thanks a lot. 👍